src/v2/view-builder/views/device/SignInDeviceView.js

```diff
@@ -24,13 +25,18 @@ const Body = BaseForm.extend({
   },
 });
 
-// override the footer to add all the supported links except the sign out link
-// no session is granted at this point
-const Footer = IdentifierFooter.extend({
-  hasBackToSignInLink: false
-});
-
 export default BaseView.extend({
   Body,
-  Footer: Footer
+
+  constructor: function() {
+    BaseView.apply(this, arguments);
+
+    const withBackLink = sessionStorageHelper.getStateHandle() ||
```

Comment:
is there any reference on how this condition is constructed? Any reference
I just wanna make sure this is the right condition and won't throw any exception

Reply:
@yannongli-okta I've added comments from the jira to this [SignInDeviceView.js](https://github.com/okta/okta-signin-widget/pull/3434/files#diff-366e8db615fc3fce7f23c097a017bd6d849d5e6baf4127ba3032002852b9f1ec), refreshed the branch with the latest master, and refactored the code to be more clear, please take a look.